config_gen.py

```diff
@@ -162,6 +163,7 @@ def fake_build(project_dir, c_build_log_path, cxx_build_log_path, verbose, make_
     env_config = env.copy()
     env_config["YCM_CONFIG_GEN_CC_PASSTHROUGH"] = cc
     env_config["YCM_CONFIG_GEN_CXX_PASSTHROUGH"] = cxx
+    env_config["QMAKESPEC"] = "linux-clang"
```

Comment:
Ok, looks like I misunderstand what it was for.

There are contexts in which one might want to use libc++ instead of libstdc++ on Linux (primarily relating to using Clang for C++11/C++14 support without having GCC 4.9 installed), but libstdc++ is the default and I'm fine with letting anyone who wants to vary from that figure it out themselves.